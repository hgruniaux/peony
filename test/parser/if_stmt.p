//# RUN: peony %s

fn test() {
    if true {}
    if true {} else {}
    if true {} else if true {}
    if true {} else if true {} else {}

    if 5i32 {}
    //~^ ERROR: mismatched types; expected 'bool', found 'i32'

    if () {}
    //~^ ERROR: expected expression
}
