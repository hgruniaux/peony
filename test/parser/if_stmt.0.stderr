<source>:9:8: error: mismatched types; expected 'bool', found 'i32'
    9 |     if 5i32 {}
      |        ~~~~
<source>:12:9: error: expected expression
   12 |     if () {}
      |         ^
